runtime/compiler/runtime/ClassUnloadAssumption.cpp

```diff
@@ -559,6 +559,15 @@ void TR_RuntimeAssumptionTable::reclaimAssumptions(OMR::RuntimeAssumption **sent
       else
          {
          sentry->markForDetach();
+         if (metaData)
+            {
+            reinterpret_cast<J9JITExceptionTable*>(metaData)->runtimeAssumptionList = NULL;
+            }
+         else if (metaData = reinterpret_cast<TR::SentinelRuntimeAssumption *>(sentry)->getOwningMetadata())
```

Comment:
I'm surprised you're not hitting a warning as error for this, i.e. `-Wparentheses` which we fixed in ed3353049afcd768f4a161ac223f7b24a9882e06:

```
warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
```

Reply:
Huh, I didn't know that was a thing. I'll wait for the sanity to finish before pushing the fix.

Reply:
As expected:
https://openj9-jenkins.osuosl.org/job/Build_JDK8_x86-64_mac_Personal/53/consoleFull

```
13:51:57  /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/openj9/runtime/compiler/runtime/ClassUnloadAssumption.cpp:566:28: error: using the result of an assignment as a condition without parentheses [-Werror,-Wparentheses]
13:51:57           else if (metaData = reinterpret_cast<TR::SentinelRuntimeAssumption *>(sentry)->getOwningMetadata())
13:51:57                    ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
13:51:57  /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/openj9/runtime/compiler/runtime/ClassUnloadAssumption.cpp:566:28: note: place parentheses around the assignment to silence this warning
13:51:57           else if (metaData = reinterpret_cast<TR::SentinelRuntimeAssumption *>(sentry)->getOwningMetadata())
13:51:57                             ^
13:51:57                    (                                                                                        )
13:51:57  /Users/jenkins/workspace/Build_JDK8_x86-64_mac_Personal/openj9/runtime/compiler/runtime/ClassUnloadAssumption.cpp:566:28: note: use '==' to turn this assignment into an equality comparison
13:51:57           else if (metaData = reinterpret_cast<TR::SentinelRuntimeAssumption *>(sentry)->getOwningMetadata())
13:51:57                             ^
13:51:57                             ==
13:51:57  1 error generated.
```

Reply:
Odd gcc on linux didn't give any warning.

Reply:
> Odd gcc on linux didn't give any warning.

I think this one may not be enabled by default on GCC. It may be under `-Wall` which we don't have enabled (yet).